source/daplink/drag-n-drop/flash_decoder.c

```diff
@@ -143,7 +143,13 @@ error_t flash_decoder_get_flash(flash_decoder_type_t type, uint32_t addr, bool a
             }
         } else if (FLASH_DECODER_TYPE_TARGET == type) {
             if (g_board_info.target_cfg) {
-                flash_start_local = g_board_info.target_cfg->flash_start;
+                region_info_t * region = g_board_info.target_cfg->flash_regions;
+                for (; region->start != 0 || region->end != 0; ++region) {
+                    if (addr >= region->start &&  addr <= region->end) {
```

Comment:
write_bin call this function with addr = 0, if the flash is not starting from addr 0x0 then flashing of bin format will fail, have you tested this? originally flash_start_local used to get populated everytime, but this is not the case anymore, I think the check should be "if (kRegionIsDefault == region->flags) {"